workflows/nsistp/create_nsistp.py

```diff
@@ -0,0 +1,156 @@
+
+        port: PortChoiceList
+        vlan: Annotated[
+            int,
+            AfterValidator(validate_vlan),
+            AfterValidator(validate_vlan_not_in_use),
+        ]
+
+        divider_1: Divider
+
+        topology: Topology
+        stp_id: StpId
+        stp_description: StpDescription | None = None
+        is_alias_in: IsAlias | None = None
+        is_alias_out: IsAlias | None = None
+        expose_in_topology: bool = True
+        bandwidth: ServiceSpeed
+
+        @model_validator(mode="after")
+        def validate_is_alias_in_out(self) -> "CreateNsiStpForm":
+            validate_both_aliases_empty_or_not(self.is_alias_in, self.is_alias_out)
+            return self
+
+    user_input = yield CreateNsiStpForm
+    user_input_dict = user_input.dict()
+
+    summary_fields = [
+        "port",
+        "vlan",
+        "topology",
+        "stp_id",
+        "stp_description",
+        "is_alias_in",
+        "is_alias_out",
+        "expose_in_topology",
+        "bandwidth",
+    ]
+    yield from create_summary_form(user_input_dict, product_name, summary_fields)
+
+    return user_input_dict
+
+
+@step("Construct Subscription model")
+def construct_nsistp_model(
+    product: UUIDstr,
+    # customer_id: UUIDstr,
+    port: UUIDstr,
+    vlan: int,
+    topology: str,
+    stp_id: str,
+    stp_description: str | None,
+    is_alias_in: str | None,
+    is_alias_out: str | None,
+    expose_in_topology: bool | None,
+    bandwidth: int | None,
+) -> State:
+    nsistp = NsistpInactive.from_product_id(
+        product_id=product,
+        customer_id=str(uuid.uuid4()),
+        status=SubscriptionLifecycle.INITIAL,
+    )
+    nsistp.nsistp.topology = topology
+    nsistp.nsistp.stp_id = stp_id
+    nsistp.nsistp.stp_description = stp_description
+    nsistp.nsistp.is_alias_in = is_alias_in
+    nsistp.nsistp.is_alias_out = is_alias_out
+    nsistp.nsistp.expose_in_topology = expose_in_topology
+    nsistp.nsistp.bandwidth = bandwidth
+
+    # TODO: change to support CustomVlanRanges
+    vlan_int = int(vlan) if not isinstance(vlan, int) else vlan
+
+    nsistp_fill_sap(nsistp, port, vlan_int)
+
+    nsistp = NsistpProvisioning.from_other_lifecycle(nsistp, SubscriptionLifecycle.PROVISIONING)
+    nsistp.description = description(nsistp)
+
+    return {
+        "subscription": nsistp,
+        "subscription_id": nsistp.subscription_id,  # necessary to be able to use older generic step functions
+        "subscription_description": nsistp.description,
+    }
+
+
+@step("Create VLANs in IMS (Netbox)")
+def ims_create_vlans(subscription: NsistpProvisioning) -> State:
+    payload = build_payload(subscription.nsistp.sap, subscription)
+    subscription.nsistp.sap.ims_id = netbox.create(payload)
+
+    return {"subscription": subscription, "payloads": [payload]}
+
+
+additional_steps = begin
+
+
+@create_workflow("Create nsistp", initial_input_form=initial_input_form_generator, additional_steps=additional_steps)
```

Comment:
This seems unnecessary

```suggestion
@create_workflow("Create nsistp", initial_input_form=initial_input_form_generator)
```